\nonstopmode{}
\documentclass[letterpaper]{book}
\usepackage[times,inconsolata,hyper]{Rd}
\usepackage{makeidx}
\usepackage[utf8]{inputenc} % @SET ENCODING@
% \usepackage{graphicx} % @USE GRAPHICX@
\makeindex{}
\begin{document}
\chapter*{}
\begin{center}
{\textbf{\huge Package `pureseqtmr'}}
\par\bigskip{\large \today}
\end{center}
\begin{description}
\raggedright{}
\inputencoding{utf8}
\item[Title]\AsIs{Predict Transmembrane Helices}
\item[Version]\AsIs{0.3}
\item[Description]\AsIs{Proteins reside in either the cell plasma of in the
cell membrane. A membrane protein goes through the
membrane at least once. There are multiple ways to span this
hydrophobic layer. One common structure is the transmembrane (alpha)
helix (TMH). Given the amino acid sequence of a membrane protein, this
package predicts which parts of the protein are TMHs.}
\item[License]\AsIs{GPL-3}
\item[Encoding]\AsIs{UTF-8}
\item[LazyData]\AsIs{true}
\item[Roxygen]\AsIs{list(markdown = TRUE)}
\item[RoxygenNote]\AsIs{7.1.0}
\item[Imports]\AsIs{ggplot2, rappdirs, stringr, tibble}
\item[Suggests]\AsIs{testthat}
\item[URL]\AsIs{}\url{https://github.com/richelbilderbeek/pureseqtmr}\AsIs{}
\item[BugReports]\AsIs{}\url{https://github.com/richelbilderbeek/pureseqtmr}\AsIs{}
\item[NeedsCompilation]\AsIs{no}
\item[Author]\AsIs{Richèl J.C. Bilderbeek [aut, cre]
(<https://orcid.org/0000-0003-1107-7049>)}
\item[Maintainer]\AsIs{Richèl J.C. Bilderbeek }\email{richel@richelbilderbeek.nl}\AsIs{}
\end{description}
\Rdcontents{\R{} topics documented:}
\inputencoding{utf8}
\HeaderA{are\_tmhs}{Are the sequences TMHs?}{are.Rul.tmhs}
%
\begin{Description}\relax
Are the sequences TMHs?
\end{Description}
%
\begin{Usage}
\begin{verbatim}
are_tmhs(protein_sequences, folder_name = get_default_pureseqtm_folder())
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{protein\_sequences}] one or more protein sequences

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}
\end{ldescription}
\end{Arguments}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  sequences <- c(
    "QEKNWSALLTAVVIILTIAGNILVIMAVSLEKKLQNATNYFLM",
    "VVIILTIRGNILVIMAVSLE"
  )
  expect_equal(c(TRUE, FALSE), are_tmhs(sequences))
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{check\_pureseqtm\_installation}{Checks the installation of PureseqTM. Throws a helpful error message if incomplete, else does nothing}{check.Rul.pureseqtm.Rul.installation}
%
\begin{Description}\relax
Checks the installation of PureseqTM.
Throws a helpful error message if incomplete,
else does nothing
\end{Description}
%
\begin{Usage}
\begin{verbatim}
check_pureseqtm_installation(folder_name = get_default_pureseqtm_folder())
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Nothing
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  expect_silent(check_pureseqtm_installation())
} else {
  expect_error(check_pureseqtm_installation())
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{check\_topology}{Check if the topology is valid. Will \LinkA{stop}{stop} if not.}{check.Rul.topology}
%
\begin{Description}\relax
Check if the topology is valid.
Will \LinkA{stop}{stop} if not.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
check_topology(topology)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{topology}] the topology as a \LinkA{tibble}{tibble}
as returned by \LinkA{predict\_proteome\_topology}{predict.Rul.proteome.Rul.topology}
\end{ldescription}
\end{Arguments}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  fasta_filename <- get_example_filename("1bhaA.fasta")
  topology <- predict_proteome_topology(fasta_filename)
  expect_silent(check_topology(topology))
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{create\_pureseqtm\_files}{Create the five PureseqTM output files, by running PureseqTM.}{create.Rul.pureseqtm.Rul.files}
%
\begin{Description}\relax
Create the five PureseqTM output files, by running PureseqTM.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
create_pureseqtm_files(
  fasta_filename,
  folder_name = get_default_pureseqtm_folder(),
  temp_folder_name = tempfile(pattern = "pureseqt_")
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{fasta\_filename}] path to a FASTA file

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}

\item[\code{temp\_folder\_name}] path of a temporary folder.
The folder does not need to exist.
Files that are out in this folder are not automatically
deleted, which is not a problem, as the default
path given by \LinkA{tempdir}{tempdir} is automatically cleaned
by the operating system
\end{ldescription}
\end{Arguments}
%
\begin{Value}
full path to the files created
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{SeeAlso}\relax
use \LinkA{run\_pureseqtm}{run.Rul.pureseqtm} to received
also the parsed output
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  fasta_filename <- get_example_filename("1bhaA.fasta")
  filenames <- create_pureseqtm_files(fasta_filename)
  expect_equal(5, length(filenames))
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{create\_pureseqtm\_proteome\_file}{Create the output file of a PureseqTM proteome run}{create.Rul.pureseqtm.Rul.proteome.Rul.file}
%
\begin{Description}\relax
Create the output file of a PureseqTM proteome run
\end{Description}
%
\begin{Usage}
\begin{verbatim}
create_pureseqtm_proteome_file(
  fasta_filename,
  topology_filename = tempfile(fileext = ".top"),
  folder_name = get_default_pureseqtm_folder()
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{fasta\_filename}] path to a FASTA file

\item[\code{topology\_filename}] name of the file to save a protein's
topology to

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
the filename
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  fasta_filename <- get_example_filename("1bhaA.fasta")
  topology_filename <- create_pureseqtm_proteome_file(fasta_filename)
  expect_equal(1, length(topology_filename))
  expect_true(file.exists((topology_filename)))
  expect_equal(0, length(readLines(topology_filename)) %% 3)
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{default\_params\_doc}{This function does nothing. It is intended to inherit is parameters' documentation.}{default.Rul.params.Rul.doc}
%
\begin{Description}\relax
This function does nothing. It is intended to inherit is parameters'
documentation.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
default_params_doc(
  download_url,
  fasta_filename,
  fasta_file_text,
  folder_name,
  protein_sequence,
  protein_sequences,
  pureseqtm_filename,
  pureseqtm_result,
  pureseqtm_url,
  temp_folder_name,
  topology,
  topology_filename,
  verbose
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{download\_url}] the URL to download PureseqTM from

\item[\code{fasta\_filename}] path to a FASTA file

\item[\code{fasta\_file\_text}] text of a FASTA file

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}

\item[\code{protein\_sequence}] a protein sequence

\item[\code{protein\_sequences}] one or more protein sequences

\item[\code{pureseqtm\_filename}] filename to write the PureseqTM results to

\item[\code{pureseqtm\_result}] the result of a PureseqTM run

\item[\code{pureseqtm\_url}] URL of the PureseqTM git repository

\item[\code{temp\_folder\_name}] path of a temporary folder.
The folder does not need to exist.
Files that are out in this folder are not automatically
deleted, which is not a problem, as the default
path given by \LinkA{tempdir}{tempdir} is automatically cleaned
by the operating system

\item[\code{topology}] the topology as a \LinkA{tibble}{tibble}
as returned by \LinkA{predict\_proteome\_topology}{predict.Rul.proteome.Rul.topology}

\item[\code{topology\_filename}] name of the file to save a protein's
topology to

\item[\code{verbose}] set to TRUE for more output
\end{ldescription}
\end{Arguments}
%
\begin{Note}\relax
This is an internal function, so it should be marked with
\code{@noRd}. This is not done, as this will disallow all
functions to find the documentation parameters
\end{Note}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
\inputencoding{utf8}
\HeaderA{get\_default\_pureseqtm\_folder}{Get the path to the folder where this package installs PureseqTM by default}{get.Rul.default.Rul.pureseqtm.Rul.folder}
%
\begin{Description}\relax
Get the path to the folder where this package installs
PureseqTM by default
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_default_pureseqtm_folder()
\end{verbatim}
\end{Usage}
%
\begin{Value}
the path to the folder where this package installs
PureseqTM by default
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (rappdirs::app_dir()$os == "unix") {
  expect_true(
    grepl(
      "/home/[A-Za-z0-9_]*/.local/share",
      get_default_pureseqtm_folder()
    )
  )
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{get\_example\_filename}{Get the full path to a PureseqTM example files}{get.Rul.example.Rul.filename}
%
\begin{Description}\relax
Get the full path to a PureseqTM example files
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_example_filename(filename, folder_name = get_default_pureseqtm_folder())
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{filename}] name of the example file, without the path

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}
\end{ldescription}
\end{Arguments}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{SeeAlso}\relax
use \LinkA{get\_example\_filenames}{get.Rul.example.Rul.filenames} to get all PureseqTM
example filenames
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  expect_true(file.exists(get_example_filename("1bhaA.fasta")))
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{get\_example\_filenames}{Get the full path to all PureseqTM example files}{get.Rul.example.Rul.filenames}
%
\begin{Description}\relax
Get the full path to all PureseqTM example files
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_example_filenames(folder_name = get_default_pureseqtm_folder())
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
a character vector with all PureseqTM example files
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{SeeAlso}\relax
use \LinkA{get\_example\_filename}{get.Rul.example.Rul.filename} to get the full
path to a PureseqTM example file
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  filenames <- get_example_filenames()
  expect_true(all(file.exists(filenames)))
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{get\_pureseqtm\_url}{Get the URL of the PureseqTM source code}{get.Rul.pureseqtm.Rul.url}
%
\begin{Description}\relax
Get the URL of the PureseqTM source code
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_pureseqtm_url()
\end{verbatim}
\end{Usage}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

url <- get_pureseqtm_url()
expect_equal(1, length(url))
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{install\_pureseqtm}{Install PureseqTM to a local folder}{install.Rul.pureseqtm}
%
\begin{Description}\relax
Install PureseqTM to a local folder
\end{Description}
%
\begin{Usage}
\begin{verbatim}
install_pureseqtm(
  folder_name = get_default_pureseqtm_folder(),
  pureseqtm_url = get_pureseqtm_url()
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}

\item[\code{pureseqtm\_url}] URL of the PureseqTM git repository
\end{ldescription}
\end{Arguments}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_on_travis() && !is_pureseqtm_installed()) {
  install_pureseqtm()
  expect_true(is_pureseqtm_installed())
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{is\_on\_travis}{Determines if the environment is Travis CI}{is.Rul.on.Rul.travis}
%
\begin{Description}\relax
Determines if the environment is Travis CI
\end{Description}
%
\begin{Usage}
\begin{verbatim}
is_on_travis()
\end{verbatim}
\end{Usage}
%
\begin{Value}
TRUE if run on Travis CI, FALSE otherwise
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
\inputencoding{utf8}
\HeaderA{is\_protein\_name\_line}{Is the line of text the name of a protein, as used within a FASTA filename?}{is.Rul.protein.Rul.name.Rul.line}
%
\begin{Description}\relax
Is the line of text the name of a protein, as used within
a FASTA filename?
\end{Description}
%
\begin{Usage}
\begin{verbatim}
is_protein_name_line(line)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{line}] line of text from a FASTA filename
\end{ldescription}
\end{Arguments}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

expect_true(is_protein_name_line(">5H2A_CRIGR"))
expect_false(is_protein_name_line("5H2A_CRIGR"))
expect_false(is_protein_name_line("000001111100000"))
expect_false(is_protein_name_line(NA))
expect_false(is_protein_name_line(NULL))
expect_false(is_protein_name_line(""))
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{is\_pureseqtm\_installed}{Measure if PureseqTM is installed locally}{is.Rul.pureseqtm.Rul.installed}
%
\begin{Description}\relax
Measure if PureseqTM is installed locally
\end{Description}
%
\begin{Usage}
\begin{verbatim}
is_pureseqtm_installed(folder_name = get_default_pureseqtm_folder())
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
TRUE is PureseqTM is installed locally,
FALSE otherwise
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

is_installed <- is_pureseqtm_installed()
expect_true(is_installed == TRUE || is_installed == FALSE)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{is\_tmh}{Determine if the protein sequence contains at least one TMH.}{is.Rul.tmh}
%
\begin{Description}\relax
Determine if the protein sequence contains at least one TMH.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
is_tmh(protein_sequence, folder_name = get_default_pureseqtm_folder())
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{protein\_sequence}] a protein sequence

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}
\end{ldescription}
\end{Arguments}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  expect_true(is_tmh("QEKNWSALLTAVVIILTIAGNILVIMAVSLEKKLQNATNYFLM"))
  expect_false(is_tmh("VVIILTIRGNILVIMAVSLE"))
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{is\_topology\_line}{Is the line of text the topology, as used within a FASTA filename?}{is.Rul.topology.Rul.line}
%
\begin{Description}\relax
Is the line of text the topology, as used within
a FASTA filename?
\end{Description}
%
\begin{Usage}
\begin{verbatim}
is_topology_line(line)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{line}] line of text from a FASTA filename
\end{ldescription}
\end{Arguments}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

expect_true(is_topology_line("000010101011"))
expect_false(is_topology_line(">5H2A_CRIGR"))
expect_false(is_topology_line("5H2A_CRIGR"))
expect_false(is_topology_line(NA))
expect_false(is_topology_line(NULL))
expect_false(is_topology_line(""))
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{plot\_topology}{Plot the topology}{plot.Rul.topology}
%
\begin{Description}\relax
Plot the topology
\end{Description}
%
\begin{Usage}
\begin{verbatim}
plot_topology(topology)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{topology}] the topology as a \LinkA{tibble}{tibble}
as returned by \LinkA{predict\_proteome\_topology}{predict.Rul.proteome.Rul.topology}
\end{ldescription}
\end{Arguments}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
\inputencoding{utf8}
\HeaderA{predict\_proteome\_topology}{Predict the topology of a proteome}{predict.Rul.proteome.Rul.topology}
%
\begin{Description}\relax
Predict the topology of a proteome
\end{Description}
%
\begin{Usage}
\begin{verbatim}
predict_proteome_topology(
  fasta_filename,
  folder_name = get_default_pureseqtm_folder(),
  topology_filename = tempfile(fileext = ".top")
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{fasta\_filename}] path to a FASTA file

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}

\item[\code{topology\_filename}] name of the file to save a protein's
topology to
\end{ldescription}
\end{Arguments}
%
\begin{Value}
a \LinkA{tibble}{tibble}
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  fasta_filename <- get_example_filename("1bhaA.fasta")
  topology <- predict_proteome_topology(fasta_filename)
  expect_true("name" %in% names(topology))
  expect_true("topology" %in% names(topology))
  expect_equal(1, nrow(topology))
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{predict\_topology\_from\_sequence}{Run PureseqTM directy on a protein sequence}{predict.Rul.topology.Rul.from.Rul.sequence}
%
\begin{Description}\relax
Run PureseqTM directy on a protein sequence
\end{Description}
%
\begin{Usage}
\begin{verbatim}
predict_topology_from_sequence(
  protein_sequence,
  folder_name = get_default_pureseqtm_folder()
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{protein\_sequence}] a protein sequence, with
the amino acids as capitals, for
example \code{MEILCEDNTSLSSIPNSL}

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
a topology as a string of zeroes and ones
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  protein_sequence <- paste0(
    "QEKNWSALLTAVVIILTIAGNILVIMAVSLEKKLQNATNYFLM",
    "SLAIADMLLGFLVMPVSMLTILYGYRWP"
  )
  topology <- predict_topology_from_sequence(protein_sequence)
  expect_true(is_topology_line(topology))
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{pureseqtmr}{pureseqtmr: estimate the topoplogy of membrane proteins}{pureseqtmr}
%
\begin{Description}\relax
Proteins reside in either the cell plasma of in the
cell membrane. A membrane protein goes through the
membrane at least once. There are multiple ways to span this
hydrophobic layer. One common structure is the transmembrane (alpha)
helix (TMH). Given the amino acid sequence of a membrane protein, this
package predicts which parts of the protein are TMHs
\end{Description}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  # Obtain an example filename
  fasta_filename <- get_example_filename("1bhaA.fasta")

  # Get the topology as a tibble
  topology <- predict_proteome_topology(fasta_filename)
  expect_true("name" %in% names(topology))
  expect_true("topology" %in% names(topology))
  expect_equal(1, nrow(topology))

  # show the topology
  plot_topology(topology)
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{run\_pureseqtm}{Runs PureseqTM for one gene and returns the parsed results}{run.Rul.pureseqtm}
%
\begin{Description}\relax
Runs PureseqTM for one gene and returns the parsed results
\end{Description}
%
\begin{Usage}
\begin{verbatim}
run_pureseqtm(
  fasta_filename,
  folder_name = get_default_pureseqtm_folder(),
  temp_folder_name = tempfile(pattern = "pureseqt_")
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{fasta\_filename}] path to a FASTA file

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}

\item[\code{temp\_folder\_name}] path of a temporary folder.
The folder does not need to exist.
Files that are out in this folder are not automatically
deleted, which is not a problem, as the default
path given by \LinkA{tempdir}{tempdir} is automatically cleaned
by the operating system
\end{ldescription}
\end{Arguments}
%
\begin{Value}
full path to the files created
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{SeeAlso}\relax
\begin{itemize}

\item{} 
Use \LinkA{create\_pureseqtm\_files}{create.Rul.pureseqtm.Rul.files} to only create the PureseqTM
output files

\item{} 
Use \LinkA{run\_pureseqtm\_proteome}{run.Rul.pureseqtm.Rul.proteome} to run PureseqTM on multiple genes


\end{itemize}

\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  fasta_filename <- get_example_filename("1bhaA.fasta")
  topology_text <- run_pureseqtm_proteome(fasta_filename)
  proteome_text <- readLines(fasta_filename)
  expect_equal(3, length(topology_text))
  expect_equal(proteome_text[1], topology_text[1])
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{run\_pureseqtm\_proteome}{Run PureseqTM on a proteome}{run.Rul.pureseqtm.Rul.proteome}
%
\begin{Description}\relax
Run PureseqTM on a proteome
\end{Description}
%
\begin{Usage}
\begin{verbatim}
run_pureseqtm_proteome(
  fasta_filename,
  folder_name = get_default_pureseqtm_folder(),
  topology_filename = tempfile(fileext = ".top")
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{fasta\_filename}] path to a FASTA file

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}

\item[\code{topology\_filename}] name of the file to save a protein's
topology to
\end{ldescription}
\end{Arguments}
%
\begin{Value}
the topology of the proteome, using the same output as PureseqTM.
Use \LinkA{predict\_proteome\_topology}{predict.Rul.proteome.Rul.topology} to get the topology as a
\end{Value}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{SeeAlso}\relax
\begin{itemize}

\item{} 
Use \LinkA{predict\_proteome\_topology}{predict.Rul.proteome.Rul.topology} to predict the topology
of a proteome

\item{} 
Use \LinkA{create\_pureseqtm\_files}{create.Rul.pureseqtm.Rul.files} to only create the PureseqTM
output files

\item{} 
Use \LinkA{run\_pureseqtm}{run.Rul.pureseqtm} to run PureseqTM on one gene in more detail


\end{itemize}

\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  fasta_filename <- get_example_filename("1bhaA.fasta")
  topology <- run_pureseqtm_proteome(fasta_filename)

  expect_true(is_protein_name_line(topology[1]))

  # Second line is the protein's amino acid sequence
  expect_equal(
    topology[2],
    paste0(
      "QAQITGRPEWIWLALGTALMGLGTLYFLVKGMGVS",
      "DPDAKKFYAITTLVPAIAFTMYLSMLLGYGLTMVPF"
    )
  )
  expect_true(is_topology_line(topology[3]))
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{run\_pureseqtm\_to\_file}{Creates a FASTA-like file, that has the locations of the amino acids.}{run.Rul.pureseqtm.Rul.to.Rul.file}
%
\begin{Description}\relax
Creates a FASTA-like file, that has the locations
of the amino acids.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
run_pureseqtm_to_file(
  fasta_filename,
  pureseqtm_filename,
  folder_name = get_default_pureseqtm_folder()
)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{fasta\_filename}] path to a FASTA file

\item[\code{pureseqtm\_filename}] filename to write the PureseqTM results to

\item[\code{folder\_name}] superfolder of PureseqTM.
The superfolder's name is \code{/home/[user\_name]/.local/share}
by default, as can be obtained by
\LinkA{get\_default\_pureseqtm\_folder}{get.Rul.default.Rul.pureseqtm.Rul.folder}
\end{ldescription}
\end{Arguments}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  pureseqtm_filename <- tempfile()
  run_pureseqtm_to_file(
    fasta_filename = get_example_filename("1bhaA.fasta"),
    pureseqtm_filename = pureseqtm_filename
  )
  expect_true(file.exists(pureseqtm_filename))
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{tally\_tmhs}{Count the number of TMHs in a topology}{tally.Rul.tmhs}
%
\begin{Description}\relax
Count the number of TMHs in a topology
\end{Description}
%
\begin{Usage}
\begin{verbatim}
tally_tmhs(topology)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{topology}] the topology as a \LinkA{tibble}{tibble}
as returned by \LinkA{predict\_proteome\_topology}{predict.Rul.proteome.Rul.topology}
\end{ldescription}
\end{Arguments}
%
\begin{Value}
a \LinkA{tibble}{tibble} with the number of
TMHs per protein
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
library(testthat)

if (is_pureseqtm_installed()) {
  topology <- predict_proteome_topology(
    get_example_filename("1bhaA.fasta")
  )
  tally <- tally_tmhs(topology)
  expect_true("name" %in% names(tally))
  expect_true("n_tmhs" %in% names(tally))
  expect_equal(nrow(topology), nrow(tally))
  expect_equal(1, nrow(tally))
  expect_equal(2, tally$n_tmhs[1])
}
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{uninstall\_pureseqtm}{Uninstall PureseqTM}{uninstall.Rul.pureseqtm}
%
\begin{Description}\relax
Uninstall PureseqTM
\end{Description}
%
\begin{Usage}
\begin{verbatim}
uninstall_pureseqtm(folder_name = get_default_pureseqtm_folder())
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{folder\_name}] name of the folder
where the PureseqTM files are installed.
The name of the PureseqTM binary file will be at
\code{[folder\_name]/PureseqTM\_Package}
\end{ldescription}
\end{Arguments}
%
\begin{Author}\relax
Richèl J.C. Bilderbeek
\end{Author}
\printindex{}
\end{document}
